## Week 6 Dataset

### Objectives:
    1. Find out how many countries are there in the dataset.
    2. Find out how many countries with respect to the region are available in the dataset
    3. On average which region has the highest
        a. GDP per capita (current US$)
       b. International trade: Imports (million US$)
    4. Find out which region is more consistent with respect to GDP per capita (current US$)

### Importing relevant Libraries

In [1]:
import numpy as np
import pandas as pd


### Reading the dataset

In [2]:
df = pd.read_excel('Data\Dataset_w6.xlsx')
df.head()

country          Region  Surface area (km2)  \
0          Afghanistan    SouthernAsia              652864   
1              Albania  SouthernEurope               28748   
2              Algeria  NorthernAfrica             2381741   
3               Angola    MiddleAfrica             1246700   
4  Antigua and Barbuda       Caribbean                 442   

   Population in thousands (2017)  GDP per capita (current US$)  \
0                           35530                         623.2   
1                            2930                        3984.2   
2                           41318                        4154.1   
3                           29784                        4714.1   
4                             102                       14764.5   

   Agricultural production index   International trade: Imports (million US$)  
0                             125                                        3568  
1                             134                                        4669  
2                             160                                       47091  
3                             175                                        8790  
4                              88                                         491

### 1. Number of countries in the dataset

In [3]:
df['country'].count()

196

There are 196 countries in the dataset

### 2. Number of countries with respect to region

In [4]:
df['Region'].value_counts()

WesternAsia          18
EasternAfrica        17
Caribbean            16
WesternAfrica        15
SouthAmerica         12
SouthernEurope       12
South-easternAsia    11
NorthernEurope       10
EasternEurope        10
MiddleAfrica          9
SouthernAsia          9
CentralAmerica        8
EasternAsia           7
WesternEurope         7
NorthernAfrica        5
Melanesia             5
Polynesia             5
CentralAsia           5
SouthernAfrica        5
NorthernAmerica       4
Micronesia            4
Oceania               2
Name: Region, dtype: int64

There are 22 regions in the dataset with WesternAsia having largest number of countries and Oceania having least number of countries

### 3. (a) Highest average G.D.P by region

In [40]:
df[['Region', 'GDP per capita (current US$)']].groupby(by='Region').mean().sort_values(by='GDP per capita (current US$)',ascending=False).head(1)

GDP per capita (current US$)
Region                                       
NorthernAmerica                       57659.0

NorthernAmerica has the highest G.D.P per capita among the 22 regions.

### (b). Highest average international imports

In [18]:
df[['Region', 'International trade: Imports (million US$)']].groupby(by='Region').mean().sort_values(by='International trade: Imports (million US$)',ascending=False).head(1)

International trade: Imports (million US$)
Region                                                     
NorthernAmerica                                   663552.25

Northern America has the highest import among all the regions

### 4. Region with least variation in G.D.P

In [38]:
df[['Region', 'GDP per capita (current US$)']].groupby(by='Region').std().sort_values(by='GDP per capita (current US$)', ascending=True).head(1)

GDP per capita (current US$)
Region                                     
WesternAfrica                    815.095399

Western African countries has least variation in G.D.P per capita among them.